configs/example/gem5_library/x86-global-inst-tracker.py

```diff
@@ -0,0 +1,143 @@
+from gem5.components.cachehierarchies.classic.private_l1_cache_hierarchy import (
+    PrivateL1CacheHierarchy,
+)
+from gem5.components.memory.single_channel import SingleChannelDDR4_2400
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.components.processors.simple_processor import SimpleProcessor
+from gem5.isas import ISA
+from gem5.resources.resource import obtain_resource
+from gem5.simulate.exit_event import ExitEvent
+from gem5.simulate.simulator import Simulator
+
+"""
+
+Usage:
+
+gem5.opt configs/example/gem5_library/x86-global-inst-tracker.py
+
+"""
+
+cache_hierarchy = PrivateL1CacheHierarchy(
+    l1d_size="64kB",
+    l1i_size="64kB",
+)
+
+memory = SingleChannelDDR4_2400("1GB")
+
+processor = SimpleProcessor(cpu_type=CPUTypes.TIMING, num_cores=9, isa=ISA.X86)
+
+# setup instruction tracker
+
+# first, we need to create a global instruction tracker
+global_inst_tracker = GlobalInstTracker(
+    # threshold to trigger the event
+    inst_threshold=100_000_000
+)
+
+# then, we create a local instruction tracker for each core
+# and store them in a list
+all_trackers = []
+
+for core in processor.get_cores():
+    tracker = LocalInstTracker(
+        # we pass in the glboal instruction tracker to the local one
+        global_inst_tracker=global_inst_tracker,
+        # this parameter tells the tracker to start listening to instructions
+        # from the beginning of the simulation. If set to False, the tracker
+        # will
+        start_listening=True,
+    )
+    # we attach the tracker to the core
+    core.core.probeListener = tracker
+    # then store the tracker in the list
+    all_trackers.append(tracker)
+
+
+board = SimpleBoard(
+    clk_freq="1GHz",
+    processor=processor,
+    memory=memory,
+    cache_hierarchy=cache_hierarchy,
+)
+
+board.set_se_binary_workload(
+    binary=obtain_resource(resource_id="x86-matrix-multiply-omp"),
+    arguments=[100, 8],
+)
+
+
+def max_inst_handler():
+    # this handler tests the instruction tracker
+    # when it reached this function, it means that it successfully raised
+    # the ExitEvent.MAX_INSTS event
+    print("Reached MAX_INSTS with 100000000 instructions")
+    print("Changing threshold to 20000")
+    # we can change the threshold of the global instruction tracker
+    global_inst_tracker.changeThreshold(20000)
+    # we need to reset the counter of the global instruction tracker
+    # the counter does not reset automatically
+    global_inst_tracker.resetCounter()
+    m5.stats.dump()
+    m5.stats.reset()
+    yield False
+    print("Reached MAX_INSTS with 20000 instructions")
+    print("Stop listening to instructions")
+    # we can stop listening to instructions
+    for tracker in all_trackers:
+        tracker.stopListening()
+    """
+    similarly, we can start listening to instructions again by calling:
+
+    for tracker in all_trackers:
+        tracker.startListening()
+
+    """
```

Comment:
Don't do multiline comments with `"""`. `"""` should only be used for doc string.